python/dgl/data/movielens.py

```diff
@@ -0,0 +1,640 @@
+        Testing graph
+    user_feat: torch.Tensor
+        User features
+    movie_feat: torch.Tensor
+        Movie features
+    info: dict
+        Store training, validation and testing rating pairs.
+
+    Notes
+    -----
+    - The number of edges is doubled to form an undirected(bidirected) graph structure.
+    - Each time when the dataset is loaded with a different setting of training, validation and testing ratio, the param
+    :obj:`"force_reload"` should be set to :obj:`"True"`. Otherwise the previous split of dataset will be loaded and not be
+    updated in the storage.
+
+    Examples
+    --------
+    >>> from dgl.data import MovieLensDataset
+    >>> 
+    >>> dataset = MovieLensDataset(name='ml-100k', valid_ratio=0.2)
+    >>> train_g, valid_g, test_g = dataset[0]
+    >>> train_g
+    Graph(num_nodes=2625, num_edges=128000,
+        ndata_schemes={}
+        edata_schemes={'_ID': Scheme(shape=(), dtype=torch.int64), 'etype': Scheme(shape=(), dtype=torch.float32)})
+    >>>
+    >>> # get ratings of edges in the training graph
+    >>> ratings = train_g.edata['etype']
+    >>> ratings
+    tensor([3., 3., 2.,  ..., 4., 4., 2.])
+    >>>
+    >>> # get training, validation and testing rating pairs
+    >>> train_rating, valid_rating, test_rating = \
+    ...     dataset.info['train_rating_pairs'], dataset.info['valid_rating_pairs'], dataset.info['test_rating_pairs']
+    >>> train_rating[0] # node index of users in training rating pairs
+    tensor([614, 772, 531,  ..., 674, 639, 740])
+    >>> train_rating[1] # node index of movies in training rating pairs
+    tensor([1236,  954, 1487,  ..., 1842, 1631, 1168])
+    >>>
+    >>> # get the rating of a certain user-movie rating pair
+    >>> u, m = train_rating[0][0], train_rating[1][0]
+    >>> eid = train_g.edge_ids(u, m)
+    >>> rating = train_g.edata['etype'][eid]
+    >>> rating
+    tensor([3.])
+    >>>
+    >>> # get input features of users and movies respectively
+    >>> user_feat, movie_feat = \
+    ...     dataset.user_feat, dataset.movie_feat
+    >>> user_feat
+    tensor([[0.4800, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
+            [1.0600, 1.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
+            [0.4600, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
+            ...,
+            [0.4000, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
+            [0.9600, 1.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000],
+            [0.4400, 0.0000, 0.0000,  ..., 0.0000, 0.0000, 0.0000]])
+
+    """
+
+    _url = {
+        "ml-100k": "http://files.grouplens.org/datasets/movielens/ml-100k.zip",
+        "ml-1m": "http://files.grouplens.org/datasets/movielens/ml-1m.zip",
+        "ml-10m": "http://files.grouplens.org/datasets/movielens/ml-10m.zip",
+    }
+
+    def __init__(
+        self,
+        name,
+        valid_ratio,
+        test_ratio=None,
+        raw_dir=None,
+        force_reload=None,
+        verbose=None,
+        transform=None,
```

Comment:
Since data splitting is random, add an optional argument for fixing the random seed.

Reply:
fix